cmhw/org/cyanogenmod/hardware/HighTouchSensitivity.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2015 The CyanogenMod Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.cyanogenmod.hardware;
+
+import java.io.File;
+import org.cyanogenmod.hardware.util.FileUtils;
+import android.os.SystemProperties;
+
+/**
+ * Glove mode / high touch sensitivity
+ */
+public class HighTouchSensitivity {
+
+    private static String GLOVE_PATH = "/sys/devices/i2c-3/3-0024/main_ttsp_core.cyttsp4_i2c_adapter/signal_disparity";
+
+    /**
+     * Whether device supports high touch sensitivity.
+     *
+     * @return boolean Supported devices must return always true
+     */
+    public static boolean isSupported() {
+        File f = new File(GLOVE_PATH);
+        SystemProperties.set("hw.glove.mod.perm.w.on","1");
+        return f.exists();
+    }
+
+    /**
+     * This method return the current activation status of high touch sensitivity
+     *
+     * @return boolean Must be false if high touch sensitivity is not supported or not activated,
+     * or the operation failed while reading the status; true in any other case.
+     */
+    public static boolean isEnabled() {
+        int i;
+        i = Integer.parseInt(FileUtils.readOneLine(GLOVE_PATH));
+
+        return i == 0 ? true : false;
+    }
+
+    /**
+     * This method allows to setup high touch sensitivity status.
+     *
+     * @param status The new high touch sensitivity status
+     * @return boolean Must be false if high touch sensitivity is not supported or the operation
+     * failed; true in any other case.
+     */
+    public static boolean setEnabled(boolean status) {
+        SystemProperties.set("hw.glove.mod.perm.w.on","0");
```

Comment:
Hi shouldn't you be getting rather than setting sys property also please change to bool